### DISCUSSION TITLE
RDA not considering the last environmental variable

### DISCUSSION THREAD

**Discussion by MinervaNunez** [vegandevs/vegan#582] [Q&A]
Hello,

I've been using the ada function to plot an rda with 8 environmental variables and 24 observations. At first it worked perfectly, but now every time that I run the code, it never considers the last variable I place at the end of the formula. I don't know why this is happening. I know is not a matter of my files or the data frame since all the variables have a result when I place them in different order, it is always the last environmental variable that never appears on the RDA calculations.

This is my code: my.rda.810<-rda(spec.df.810 ~ace+pro+but+ace.pro+ace.but+pro.but+ace.pro.but+ctr,env.810)
In this example, the last environmental variable (ctr), won't appear in the RDA calculations.

I would appreciate any leads regarding this because I haven't seen anyone having a similar issue.

**Comment by gavinsimpson** [comment]
What does the output generated when you print `my.rda.810` show? I suspect somewhere in the output is a note about some variables being *aliased*. This means that one or more columns of the model matrix are linearly dependent on one or more other columns in the matrix. In other words, you might have 10 variables but you don't have 10 unique pieces of information.

But do edit your question to include the output when you print your RDA object.

**Comment by jarioksa** [comment]
We have seen this problem several times. Two current-ish issues in github are #511 and #456, but there are others – and many more in StackOverflow. In all cases so far the reason has been the same: the one that @gavinsimpson explained.
